test-result-summary-client/src/Dashboard/Widgets/Graph/ODM.jsx

```diff
@@ -80,14 +80,13 @@ export default class ODM extends Component {
 
         // combine results having the same JDK build date
         results.forEach(( t, i ) => {
-            if ( t.buildResult !== "SUCCESS" ) return;
-
+            if ( t.buildResult !== "SUCCESS" || ! Array.isArray(t.aggregateInfo) || t.aggregateInfo.length < 1 || ! t.aggregateInfo[0].jdkDate || ! Array.isArray(t.aggregateInfo[0].metrics) || t.aggregateInfo[0].metrics.length < 1 || ! t.aggregateInfo[0].metrics[0]["value"] || Object.keys(t.aggregateInfo[0].metrics[0]["value"]).length === 0) return;
             // TODO: current code only considers one interation. This needs to be updated
-            if ( t.tests[0].testData && t.tests[0].testData.metrics && t.tests[0].testData.metrics.length > 0 ) {
-                const jdkDate = t.tests[0].jdkDate;
-                if ( !t.tests[0].testData.metrics[0].value
-                    || t.tests[0].testData.metrics[0].value.length === 0
-                    || t.tests[0].testData.metrics[0].name !== "Global Throughput" ) {
+            for ( let aggregateInfo of t.aggregateInfo ) {
+                const jdkDate = aggregateInfo.jdkDate;
+                if ( !Array.isArray(aggregateInfo.metrics)
+                    || aggregateInfo.metrics[0].value.length === 0
+                    || aggregateInfo.metrics[0].name !== "Global Throughput" ) {

```

Comment:
The concerned metric `Global Throughput` doesn't need to be under index 0 (i.e. `aggregateInfo.metrics[0].name`).